src/main/java/com/browserstack/automate/ci/jenkins/local/JenkinsBrowserStackLocal.java

```diff
@@ -51,7 +51,7 @@ public JenkinsBrowserStackLocal(String accesskey, LocalConfig localConfig, Strin
             arguments.add(args[i]);
         }
 
-        localIdentifier = UUID.randomUUID().toString().replaceAll("\\-", "") + "-" + buildTag.replaceAll("\\s","").replaceAll("#", "_");
+        localIdentifier = (UUID.randomUUID().toString() + "-" + buildTag).replaceAll("[^a-bA-Z0-9_\\-\\.]", "_");
```

Comment:
why not simple \w

https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html

Reply:
Yes, good call. I'll change that.